posts/2023-11-05-telegram-bot-for-file-sharing-using-quarkus.adoc

```diff
@@ -0,0 +1,495 @@
+
+All we need to do is to add a few properties to `application.properties` file:
+[source]
+----
+quarkus.mongodb.devservices.enabled=true
+quarkus.mongodb.devservices.image-name=mongo
+quarkus.mongodb.devservices.port=27018
+
+quarkus.minio.devservices.enabled=true
+quarkus.minio.devservices.image-name=minio/minio
+quarkus.minio.devservices.access-key=testtest
+quarkus.minio.devservices.secret-key=testtest
+quarkus.minio.devservices.port=9000
+----
+
+After we run our application in dev mode, Quarkus will up the specified containers automatically and create MinioClient with corresponding properties.
+
+Well, our backend application is ready to up. Being in the root directory run:
+```bash
+quarkus dev
+```
+
+
+== Frontend
+For the frontend application, we will use https://docs.quarkiverse.io/quarkus-quinoa/dev/[Quarkus Quinoa project] using React with TypeScript.
+All possible framework combinations you can see on https://docs.quarkiverse.io/quarkus-quinoa/dev/web-frameworks.html[Quarkus Quinoa - Web Frameworks]
+
+Inside our generated application we can see the directory  `src/main/webui` - it’s the root for our NodeJS application with  a`package.json` file. Let's remove the default implementation by removing the whole directory `src/main/webui`.  Then place yourself in `file-sharing-bot/src/main` and execute:
+```bash
+npx create-react-app webui --template typescript
+```
+(it requires https://nodejs.org/en/download/package-manager#alternatives-2[node] and https://www.npmjs.com/package/npx[npx] to be already installed).
+
+In the directory `src/main/webui` we can see the new generated project.
+
+As usual, let's add `bootstrap`. Being in `/frontend/src/main/webui` execute:
+```bash
+npm install bootstrap
+```
+
+The last thing we need is to change the `webui/src/App.tsx` file. We have to add one button and the function that sends an HTTP request to our backend application when the `onClick()` event is triggered. We will also retrieve the chatId variable from the query parameter:
+[source,javascript]
+----
+import React from 'react';
+import logo from './logo.svg';
+import './App.css';
+import '../node_modules/bootstrap/dist/css/bootstrap.css';
+
+function App() {
+  const queryParams = new URLSearchParams(window.location.search);
+  const chatId = queryParams.get('chatId');
+
+  function download() {
+    window.location.href = `http://127.0.0.1:8091/api/v1/file-sharing-bot/download?chatId=${chatId}`;
+  }
+
+  return (
+      <div className="App">
+        <h1 className="display-1"> Click on the button to download</h1>
+        <button className="btn btn-primary" onClick={() => download()}>Download</button>
+      </div>
+  );
+}
+
+export default App;
+
+----
+
+Yes, hardcode it (3).
+
+And just in case, change the default port for our application by adding the property `quarkus.http.port=8091` to `application.properties`
```

Comment:
```suggestion
To avoid port conflicts, change the default port for our application by adding the property `quarkus.http.port=8091` to `application.properties`
```